### ISSUE TITLE
Drop support for QUIP and PEXSI

### ISSUE THREAD

**Issue by oschuett** [cp2k/cp2k#3424]
I think we should drop support for QUIP and PEXSI because for all I can tell they are not really being used. 

While in the past the maintenance effort was rather low, we would now have to invest quite a bit of work to migrate them to CMake and Spack.

Are there any objections?

**Comment by fstein93** [comment]
I am in favor of discussing about it (maybe at the next developers' meeting). I have never used any of these packages but I am not an average user anyways.
A workaround for PEXSI could be to move on to PEXSI 2.0.0. According to its Manual, it does build with CMake (see [here](https://pexsi.readthedocs.io/en/latest/install.html?highlight=cmake#build-pexsi)).

**Comment by RMeli** [comment]
I'm usually in favor of removing unused/barely used code. However, I just want to point out that PEXSI is available in Spack.

In https://github.com/spack/spack/pull/36049 I added version `1.2` and `2.0`, but never got around to test them with CP2K. As @fstein93 pointed out, these two versions use CMake and we could get rid of the custom `FindPEXSI.cmake` we have.

We clearly never tested `cp2k+pexsi` since I can see that `CP2K_USE_PEXSI` is missing from the CP2K Spack package, so a bit more work will be needed there too.

QUIP, however, doesn't seem to be available in Spack, so it will need to be added if it is needed as a dependency.

**Comment by oschuett** [comment]
Thanks for chiming in. I have no doubt that we _could_ keep supporting these libraries, but there is an opportunity cost.

Patrick already concluded in [his thesis](https://dx.doi.org/10.3929%2Fethz-a-010738698) that PEXSI is essentially always outperformed by diagonalization - with the exception of extreme strong scaling:

> Currently, the main limitation for a broad application of the PEXSI approach
> to a wide range of systems is its high computational costs in the sense that
> PEXSI starts to be beneficial only at large system sizes and requires massive
> parallelisation.

He also noted that other codes fare better because of their more localized basis sets:

> The results obtained in this work can not be directly compared to the SIESTA-
> PEXSI approach because SIESTA uses numerical basis functions [45, 46] while
> CP2K is based on Gaussian-type functions.

As for QUIP, I believe its crown jewel GAP, while groundbreaking at the time, has since been superseded by more sophisticated ML potentials such as NNP, NequIP/Allegro, and DeePMD.

It's also noteworthy that both packages are only ever mentioned in the forum in the context of installation issues.

So I see very little evidence that the work we would have to put in to maintaining these packages is actually worth it.

**Comment by fstein93** [comment]
Thank you for these insights. I was not aware of them. Then, I am in favor of dropping them entirely. I suggest to communicate this decision to the users and drop them finally in the second half of the year.

**Comment by alas-go** [comment]
Dears, I am a normal user for solids and catalyst. PEXSI should still be kept in the future because the simulations for large system sizes, especially with vacumm slab are required and this focuse keeps increasing. Quip meet some challenges from  ML potentials.

**Comment by oschuett** [comment]
> PEXSI should still be kept in the future because the simulations for large system sizes, especially with vacumm slab...

Yes, metal slabs in vacuum is where one would most likely expect PEXSI to shine. Yet, AFAIK people are not using it.  Maybe you could share some publications that did?

> Quip meet some challenges from ML potentials.

Could you be a bit more specific? My understanding is that Gaussian Processes, on which GAP is based, have been largely replaced by neural networks.

**Comment by alas-go** [comment]
> > PEXSI should still be kept in the future because the simulations for large system sizes, especially with vacumm slab...
> 
>Yes, metal slabs in vacuum is where one would most likely expect PEXSI to shine. Yet, AFAIK people are not using it. Maybe you could share some publications that did?

Type kewword "vacuum slab catalyst  cp2k" and search through google scholar, it shows 1560 publications in all and 87 publications in 2024. 

PEXI could  cut down the cpu time used because it  speed up the computation in Quick Step.

Here are some typical
https://pubs.acs.org/doi/abs/10.1021/acscatal.9b05175
https://pubs.rsc.org/en/content/articlehtml/2022/aw/d1sc01201k
https://pubs.acs.org/doi/abs/10.1021/acs.jpcc.1c07040
https://pubs.acs.org/doi/abs/10.1021/acs.jpcc.6b12054
https://pubs.acs.org/doi/full/10.1021/acscatal.1c00538
https://pubs.acs.org/doi/abs/10.1021/acscatal.2c00159

> 
> > Quip meet some challenges from ML potentials.
> 
> Could you be a bit more specific? My understanding is that Gaussian Processes, on which GAP is based, have been largely replaced by neural networks.

Sorry. I mean, traditional potentials, meet some challenges from ML  potentials and potential with neural networks. this is from several publictians i have read but i did not record them. 
Sorry, I am not expertized in this field.

**Comment by oschuett** [comment]
> Here are some typical

Thanks, but it seems none of these publications used PEXSI, which is exactly my point. Searching for "cp2k pexsi" does not yield a single application paper, although the method has been available for almost 10 years. 

> PEXI could cut down the cpu time used because it speed up the computation in Quick Step.

I'm sorry, but I have not seen any evidence to support that.

As for QUIP, thanks to #3420 one should still be able to use it like this: CP2K-> i-PI -> ASE -> QUIP.

**Comment by alas-go** [comment]
> > Here are some typical
> 
> Thanks, but it seems none of these publications used PEXSI, which is exactly my point. Searching for "cp2k pexsi" does not yield a single application paper, although the method has been available for almost 10 years.
> 
> > PEXI could cut down the cpu time used because it speed up the computation in Quick Step.
> 
> I'm sorry, but I have not seen any evidence to support that.
> 
This is based on my calculating practices.  Maybe there are other codes, which could replace PEXI.

**Comment by alas-go** [comment]
Will you update the version of PEXSI?

**Comment by fstein93** [comment]
Probably yes but not within the next release in a few days.

**Comment by Growl1234** [comment]
Looks like this issue is closed with only QUIP removed, so I'm a bit confused. Is removing PEXSI support still a plan?

**Comment by DCM-Uni-Paderborn** [comment]
> Looks like this issue is closed with only QUIP removed, so I'm a bit confused. Is removing PEXSI support still a plan?

I would opt to keep PEXI for a little bit longer. We are working on refactoring LS_SCF for a while now and on that occasion would like to natively implement an alternative method of us (https://arxiv.org/pdf/1206.5382). If this turns our to be competitive, we can drop PEXI by then ...

**Comment by oschuett** [comment]
Since Matthias managed to [upgrade to PEXSI 2](https://github.com/cp2k/cp2k/pull/4203), the primary reason for removing went await. So, for the time being it's [deprecated](https://github.com/cp2k/cp2k/blob/c2d535141ab33ca218ea2cf57485026a2fa4bd6a/src/input_cp2k_transport.F#L432).
